### PR TITLE
feat(karthik_model): neural network training on NBA CSV data

### PR DESCRIPTION
**Pull request by Kravi001**
## What I changed
- Added `karthik_model.py` in `src/mini_nn/` to train a neural network on NBA CSV data.
- Loads data from `Data/PlayerStatistics.csv`, preprocesses features, handles NaNs, and manages class imbalance.
- Trains a model to predict whether a player scores ≥20 points and saves predictions (gitignored).

## Why I changed it
- Provides a fully working neural network pipeline for NBA CSV data.
- Allows generating predictions without connecting to a database.

## How did I test it
- Ran the script locally on the full dataset (1,655,736 rows, 19 features).
- Observed class imbalance: 0 → 0.8699, 1 → 0.1301; positive class weight = 6.69.
- Training metrics:
  - Epoch 10 | Train Loss: 0.1343 | Val F1: 0.9847
  - Epoch 20 | Train Loss: 0.0942 | Val F1: 0.9400
  - Epoch 30 | Train Loss: 0.0690 | Val F1: 0.9919
  - Epoch 40 | Train Loss: 0.0624 | Val F1: 0.9946
  - Epoch 50 | Train Loss: 0.0418 | Val F1: 0.9990
  - Epoch 60 | Train Loss: 0.0330 | Val F1: 0.9997
  - Epoch 70 | Train Loss: 0.0264 | Val F1: 1.0000
- Early stopping triggered at epoch 72.
- Test results:
  - Best Epoch: 57
  - Test Accuracy: 0.9999
  - Test F1 Score: 0.9997
  - Test ROC-AUC: 1.0000

### CODE REVIEW

**Review by JonathanPLev — CHANGES_REQUESTED** [review]
can you add your results? curious to know how well this model did.

edit: sorry i see it actually. how are you getting 100% test accuracy? I think you are having leaking or overfit data.

File: src/mini_nn/karthik_model.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+    class_weights = compute_class_weight(class_weight='balanced', classes=classes, y=y)
+    pos_weight = torch.tensor([class_weights[1] / class_weights[0]], dtype=torch.float)
+    print(f"Class weight for positive class: {pos_weight.item():.2f}")
+
+    # -------------------------
+    # SPLIT
+    # -------------------------
+    X_temp, X_test, y_temp, y_test = train_test_split(X, y, test_size=0.15, random_state=42)
+    X_train, X_val, y_train, y_val = train_test_split(X_temp, y_temp, test_size=0.176, random_state=42)
+
```
Comment (JonathanPLev) [inline src/mini_nn/karthik_model.py:115]: you are definitely having leaking data here because you are having random split

File: src/mini_nn/karthik_model.py [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+    # -------------------------
+    classes = np.unique(y)
+    class_weights = compute_class_weight(class_weight='balanced', classes=classes, y=y)
+    pos_weight = torch.tensor([class_weights[1] / class_weights[0]], dtype=torch.float)
+    print(f"Class weight for positive class: {pos_weight.item():.2f}")
+
+    # -------------------------
+    # SPLIT
+    # -------------------------
+    X_temp, X_test, y_temp, y_test = train_test_split(X, y, test_size=0.15, random_state=42)
```
Comment (JonathanPLev) [inline src/mini_nn/karthik_model.py:113]: youre also not dropping any of the lines that leak data to the outcome. like youre not removing any stats like points or anything thats current game.

**Review by JonathanPLev — CHANGES_REQUESTED** [review]

**Comment by JonathanPLev** [comment]
stale
